### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by IoghiStudio**

### CODE REVIEW

File: src/averageAges.js [hunk 1]
```diff
@@ -1,61 +1,47 @@
...
- * @return {number}
- */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const momAndChild = (onlyWithSon
+    ? people.filter(person => person.sex === 'm'
+      && people.find(mother => person.mother === mother.name))
+    : people.filter(person =>
+      people.find(mother => person.mother === mother.name)));
+
```
Comment (Rijo-J) [inline src/averageAges.js:37]: For readability + code repetition, using callback helpers declared in the upper sc.
```suggestion
  const momAndChild = (
    onlyWithSon
      ? isMale(person) && isChild(person, people)
      : isChild(person, people)
  ;

```
idem for previous functions, but still works great!! 😁

**Review by Rijo-J — APPROVED** [review]
looks like hard work pays off!! 😁 🚀 passing to mentors

File: src/averageAges.js [hunk 2]
```diff
@@ -1,61 +1,47 @@
...
- * @param {object[]} people
- * @param {number} century - optional
- *
- * @return {number}
- */
+function getAverage(arr) {
+  return arr
+    .map(user => user.died - user.born)
+    .reduce((acc, user) => acc + user) / arr.length;
+}
```
Comment (naziamoff) [inline src/averageAges.js:7]: - get average what?
- what is arr?

File: src/averageAges.js [hunk 3]
```diff
@@ -1,61 +1,47 @@
...
+
 function calculateMenAverageAge(people, century) {
-  // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
+  let mans = people.filter(user => user.sex === 'm');
```
Comment (naziamoff) [inline src/averageAges.js:10]: - this comment is more grammatical, but still 😄 
```suggestion
  let men = people.filter(person => person.sex === 'm');
```

File: src/averageAges.js [hunk 4]
```diff
@@ -1,61 +1,47 @@
...
+
+  if (century !== undefined) {
+    mans = mans.filter(man => Math.ceil(man.died / 100) === century);
+  }
```
Comment (naziamoff) [inline src/averageAges.js:14]: do you need to check whether it is exactly undefined, or you just need to know whether century is falsy?

File: src/averageAges.js [hunk 5]
```diff
@@ -1,61 +1,47 @@
...
- * her as mother.
- *
- * @param {object[]} people
- * @param {boolean} withChildren - optional
- *
- * @return {number}
- */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  let womens = people.filter(user => user.sex === 'f');
```
Comment (naziamoff) [inline src/averageAges.js:20]:
```suggestion
  let women = people.filter(person => person.sex === 'f');
```

File: src/averageAges.js [hunk 6]
```diff
@@ -1,61 +1,47 @@
...
+
+  if (withChildren) {
+    womens = people
+      .filter(user => people.some(little => little.mother === user.name)
+      );
+  }
+
```
Comment (naziamoff) [inline src/averageAges.js:27]: let's filter it inside one callback

File: src/averageAges.js (hunk 1 above)
Comment (naziamoff) [reply src/averageAges.js:37]: nice point from @Rijo-J

**Review by naziamoff — CHANGES_REQUESTED** [review]
nice job
still need some polishing though

**Pull request closed without merging**